File: frontend/src/services/reviewService.js
import api from './api'

export const reviewService = {
  getReviews: (params) => api.get('/reviews', { params }),
  getSummary: () => api.get('/reviews/summary'),
  createReview: (data) => api.post('/reviews', data),
  likeReview: (id) => api.post(`/reviews/${id}/like`),
}

File: backend/src/main/java/com/restaurant/service/ReviewService.java
package com.restaurant.service;

import com.restaurant.dto.request.CreateReviewRequest;
import com.restaurant.dto.response.ReviewResponse;
import com.restaurant.dto.response.ReviewSummaryResponse;

import java.util.List;

public interface ReviewService {

    ReviewResponse createReview(CreateReviewRequest request);

    List<ReviewResponse> getReviews(Integer rating, String sort);

    ReviewSummaryResponse getSummary();

    ReviewResponse likeReview(Long id);
}
